File: app/src/main/java/torille/fi/lurkforreddit/search/SearchPresenterModule.java
package torille.fi.lurkforreddit.search;

import dagger.Module;
import dagger.Provides;
import torille.fi.lurkforreddit.data.RedditRepository;

/**
 * Created by eva on 25.4.2017.
 */
@Module
public class SearchPresenterModule {

    public SearchPresenterModule() {

    }

    @Provides
    public SearchContract.Presenter<SearchContract.View> provideSearchPresenter(RedditRepository redditRepository) {
        return new SearchPresenter(redditRepository);
    }

}

File: app/src/main/java/torille/fi/lurkforreddit/search/SearchPresenterModule.kt
package torille.fi.lurkforreddit.search

import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import torille.fi.lurkforreddit.di.scope.FragmentScoped

@Module
abstract class SearchPresenterModule {

    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun searchFragment(): SearchFragment

    @FragmentScoped
    @Binds
    abstract fun searchPresenter(searchPresenter: SearchPresenter): SearchContract.Presenter

}
